temporal-sdk/src/main/java/io/temporal/common/converter/JacksonJsonPayloadConverter.java

```diff
@@ -37,14 +37,11 @@
   private final ObjectMapper mapper;
 
   public JacksonJsonPayloadConverter() {
-    mapper = new ObjectMapper();
-    mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
-    mapper.registerModule(new JavaTimeModule());
-    mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+    this(null);
   }
 
   public JacksonJsonPayloadConverter(ObjectMapper mapper) {
-    this.mapper = mapper;
+    this.mapper = mapper != null ? mapper : newDefaultObjectMapper();
```

Comment:
I don't think that the new version of the constructors is more readable.

Reply:
These changes were required for the initial approach, I can revert them now